### DISCUSSION TITLE
cPanel

### DISCUSSION THREAD

**Discussion by hirockstar** [spencerwooo/onedrive-vercel-index#511] [Q&A] [answered]
can we deploy this on localhost???

**Comment by myl7** [comment, marked as answer]
See discussions #112 , but this is a outdated tutorial. You should change the envs to `REDIS_URL=...` only, and use `pnpm` to install dependencies in `Dockerfile`.

**Reply by hirockstar** [reply]
is there any docs of this?

**Reply by myl7** [reply]
No, as the project aims to serverless(free). But this is still a traditional Next.js project and no other extra stuff is added, so it should be no so hard to selfhost it.
